examples/platform_services/android/app/src/main/java/com/example/flutter/ExampleActivity.java

```diff
@@ -36,9 +39,18 @@ public void onMethodCall(MethodCall call, Response response) {
   }
 
   private void getBatteryLevel(Response response) {
-    BatteryManager batteryManager = (BatteryManager) getSystemService(BATTERY_SERVICE);
+    int batteryLevel = -1;
     if (VERSION.SDK_INT >= VERSION_CODES.LOLLIPOP) {
-      response.success(batteryManager.getIntProperty(BatteryManager.BATTERY_PROPERTY_CAPACITY));
+      BatteryManager batteryManager = (BatteryManager) getSystemService(BATTERY_SERVICE);
+      batteryLevel = batteryManager.getIntProperty(BatteryManager.BATTERY_PROPERTY_CAPACITY);
+    } else {
+      Intent intent = new ContextWrapper(getApplicationContext()).
+          registerReceiver(null, new IntentFilter(Intent.ACTION_BATTERY_CHANGED));
+      batteryLevel = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, -1);
```

Comment:
https://developer.android.com/training/monitoring-device-state/battery-monitoring.html says you have to divide EXTRA_LEVEL by EXTRA_SCALE to get a percentage. I guess the scale may be 1 for some (or all?) devices.